File: ios/Kwift Template/swiftResources/layouts/WeekDemoXml.swift
//
// WeekDemoXml.swift
// Created by Kwift XML
//

import UIKit
import Kwift
import MapKit

class WeekDemoXml {
    
    unowned var xmlRoot: UIView!
    func setup(_ dependency: ViewDependency) -> UIView {
        let view = LinearLayout(frame: .zero)
        view.orientation = .y
        view.padding = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
        view.gravity = .topLeft
        
        view.addSubview(
            UIWeekView(frame: .zero),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: 0, height: 0),
            margin: UIEdgeInsets(top: 0 + 0, left: 0 + 0, bottom: 0 + 0, right: 0 + 0),
            gravity: .topFill,
            weight: 1.0
        ) { view in 
            self.week = view
        }
        
        xmlRoot = view
        return view
    }
    
    unowned var week: UIWeekView!
    
}

File: android/app/src/main/java/com/lightningkite/kwifttemplate/layouts/WeekDemoXml.kt
//
// WeekDemoXml.swift
// Created by Kwift XML Android
//
package com.lightningkite.kwifttemplate.layouts

import android.view.LayoutInflater
import android.view.View
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwifttemplate.R

class WeekDemoXml {

    lateinit var week: com.lightningkite.kwift.views.android.WeekView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.week_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        week = view.findViewById<com.lightningkite.kwift.views.android.WeekView>(R.id.week)
        
        
        return view
    }
}